gpii/node_modules/gpii-oauth2/gpii-oauth2-authz-server/src/AuthServer.js

```diff
@@ -506,4 +515,24 @@ gpii.oauth2.authServer.contributeRouteHandlers = function (that, oauth2orizeServ
         }
     );
 
+    that.expressApp.post('/add-preferences',
+        function(req, res) {
+            var accessToken = gpii.oauth2.parseBearerAuthorizationHeader(req);
+            if (!accessToken) {
+                res.send(401);
+            } else {
+                var client = that.authorizationService.getClientByClientCredentialsAccessToken(accessToken);
+                if (!client) {
+                    res.send(404);
+                }
+
+                var tokenPrivs = that.authorizationService.getClientCredentialsTokenPrivs(accessToken);
+                if (!tokenPrivs.allowAddPrefs) {
+                    res.send(401);
+                } else {
+                    that.authorizationService.savePrefs(req.body, req.query.view, function (data) { res.json(data) });
```

Comment:
It looks like errors from the savePrefs operation are not propagated properly here - it should be possible to influence the status code, etc. 
There are utilities in "new Kettle" to help with this but there should be something basic here like intercepting an "isError" payload and setting a 500 statusCode or so